integration-provisioning/src/main/java/org/symphonyoss/integration/provisioning/service/UserService.java

```diff
@@ -129,6 +129,25 @@ private void createNewUser(String sessionToken, String username, String name, St
     }
   }
 
+  /**
+   * Updates user information
+   * @param sessionToken Token to access the User API.
+   * @param user User object
+   * @param name User display name
+   * @param avatar User avatar (Base64 encoded)
+   */
+  private void updateUser(String sessionToken, UserV2 user, String name, String avatar) {
+    UserAttributes userAttributes = createUserAttributes(user.getUsername(), name);
+
+    try {
+      Long userId = user.getId();
+      userApi.v1AdminUserUidUpdatePost(sessionToken, userId, userAttributes);
+      updateUserAvatar(sessionToken, userId, avatar);
+    } catch (ApiException e) {
+      throw new UpdateUserException("Failed to update user avatar", e);
```

Comment:
OK.  So if you look in updateUserAvatar it is catching ApiException already and wrapping it with a UpdateUserException("Failed to update user avatar") exactly as is done here (cut/paste?).  If that happens, it won't get caught here (because it's not a subclass of ApiException) - good.  But I assume UserApi.v1AdminUserUidUpdatePost() _does_ throw ApiException which will be caught and you are also wrapping that with UpdateUserException("Failed to update user avatar") but it is not an avatar update call that failed.

So, two things:
1. Move udpateUserAvatar() call below this try/catch since it doesn't throw the exception being caught
2. Change the error message in the exception since what failed was NOT updating the avatar.